src/compiler/scala/tools/nsc/typechecker/Infer.scala

```diff
@@ -148,7 +148,7 @@ trait Infer extends Checkable {
   }
 
   def skipImplicit(tp: Type) = tp match {
-    case mt: MethodType if mt.isImplicit  => mt.resultType
+    case mt: MethodType if mt.isImplicit  => mt.resultType //
```

Comment:
Yes, my reading of this code is that `skipImplicit` is a no-op, because `finalResultType` already strips all `MethodType`s.

Looks like `finalResultType` was introduced in https://github.com/scala/scala/commit/b9082321219f78a86e80fcea2a71d96ee4beb397#diff-62ccdfe29e58b8e9f06ed66d4cfc88a5L1086 / https://github.com/scala/scala/commit/b9082321219f78a86e80fcea2a71d96ee4beb397#diff-62ccdfe29e58b8e9f06ed66d4cfc88a5R1092